TypeCobol/Compiler/Preprocessor/AutoReplacePartialWordsTokensDocument.cs

```diff
@@ -31,11 +31,12 @@ public Token NextToken()
                 if (nextToken.TokenType == TokenType.PartialCobolWord)
                 {
                     //basic replacement mechanic, remove the ':' from the tag.
-                    //NOTE: PartialCobolWord have their ScanStateSnapshot embedded so we can use it to replace with number literal after OCCURS if need be...
+                    //NOTE: Altered token is scanned as if it was located at the beginning of the line because we only have InitialScanState here.
                     //NOTE: Does not handle '::-item' or 'item-::' partial names as '::' will turn into empty string and will produce invalid data names.
                     var originalToken = nextToken;
                     string replacedTokenText = originalToken.NormalizedText.Replace(":", string.Empty);
-                    var generatedReplacementToken = ReplaceTokensLinesIterator.GenerateReplacementToken(originalToken, replacedTokenText, _compilerOptions);
+                    var scanState = originalToken.TokensLine.InitialScanState;
```

Comment:
It works in most case to use the InitialScanState but not for syntax like:
```cobol
         05 :DVZOSM:-NB    pic 9(08).
         05 :DVZOSM:-DATA pic
           9(08) occurs 10 depending on :DVZOSM:-NB.
```
The ScanState should be recalculated like in ReplaceTokensLinesIterator.
I suggest to keep like it in this PR and wait for #2463 so we have an abstract base class to handle the correct ScanState.